snowdialect.py

```diff
@@ -572,18 +572,49 @@ def get_schema_names(self, connection, **kw):
 
         return [self.normalize_name(row[1]) for row in cursor]
 
-    def get_table_comment(self, connection, table_name, schema=None, **kw):
+    def _get_table_comment(self, connection, table_name, schema=None, **kw):
         """
-        Returns comment of table in a dictionary as described by SQLAlchemy spec
+        Returns comment of table in a dictionary as described by SQLAlchemy spec.
         """
         sql_command = "SHOW /* sqlalchemy:get_table_comment */ " \
                       "TABLES LIKE '{0}'{1}".format(
                                         table_name,
                                         (' IN SCHEMA {0}'.format(self.normalize_name(schema))) if schema else ''
                                     )
         cursor = connection.execute(sql_command)
-        ans = cursor.fetchone()
-        return {'text': ans['comment'] if ans['comment'] else None}
+        result = cursor.fetchone()
+        return result['comment']
+
+    def _get_view_comment(self, connection, table_name, schema=None, **kw):
+        """
+        Returns comment of view in a dictionary as described by SQLAlchemy spec.
+        """
+        sql_command = "SHOW /* sqlalchemy:get_view_comment */ " \
+            "VIEWS LIKE '{0}'{1}".format(
+                table_name,
+                (' IN SCHEMA {0}'.format(self.normalize_name(schema))) if schema else ''
+            )
+        cursor = connection.execute(sql_command)
+        result = cursor.fetchone()
+        return result['comment']
+
+    def get_table_comment(self, connection, table_name, schema=None, **kw):
+        """
+        Returns comment associated with a table (or view) in a dictionary as
+        SQLAlchemy expects.
+        """
+        # NOTE: Since SQLAlchemy may not (in fact, typically does not) know if
+        # this is a table or a view, we have to handle both cases here.
+        try:
+            comment = self._get_table_comment()
+            if comment is None:
```

Comment:
Note that this logic will need to be updated once you change the return value checks.